src/main/java/com/fcgl/madrid/points/service/UserActionService.java

```diff
@@ -1,22 +1,96 @@
 package com.fcgl.madrid.points.service;
 
-import com.fcgl.madrid.points.model.UserAction;
-import com.fcgl.madrid.points.repository.UserActionRepository;
+import com.fcgl.madrid.points.model.*;
+import com.fcgl.madrid.points.repository.*;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
 public class UserActionService {
     private UserActionRepository userActionRepository;
+    private ActionRepository actionRepository;
+    private TrophyRepository trophyRepository;
+    private UserTrophyRepository userTrophyRepository;
+    private UserPointsRepository userPointsRepository;
 
     @Autowired
-    public UserActionService(UserActionRepository userActionRepository) {
+    public UserActionService(UserActionRepository userActionRepository,
+                             ActionRepository actionRepository,
+                             TrophyRepository trophyRepository,
+                             UserTrophyRepository userTrophyRepository,
+                             UserPointsRepository userPointsRepository) {
         this.userActionRepository = userActionRepository;
+        this.actionRepository = actionRepository;
+        this.trophyRepository = trophyRepository;
+        this.userTrophyRepository = userTrophyRepository;
+        this.userPointsRepository = userPointsRepository;
     }
 
     public List<UserAction> findAll() {
         return userActionRepository.findAll();
     }
+
```

Comment:
TODO add `CircuitBreaker` logic. I can add this in a subsequent PR since I implemented Resilience4j. 

@AshleeCW  has a PR with some basic CircuitBreaker logic implemented in his PR for reference as to what I am talking about. https://github.com/fcgl/point-system/pull/7

Pretty much when an unexpected exception is thrown (SQL exception) a default fallback function is called